aws/Makefile

```diff
@@ -7,38 +7,90 @@ optional-generate:
 	rm -rf ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/kubeflow-shared-test-infra-poc-argo/namespaces/argo/*
+	mkdir -p ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/kubeflow-shared-test-infra-poc-argo/namespaces/argo/
 	####
 	# Add Logic to copy yaml files here
 	####
 	echo "Finish generating yaml files for kubeflow-shared-test-infra-poc-argo cluster"
 
 # Generate YAML files for Optional-Test-Infra GitOps for kubeflow-shared-test-infra-poc-tekton cluster
 optional-generate-kubeflow-shared-test-infra-poc-tekton:
+	echo ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}
 	echo "Start generating yaml files for kubeflow-shared-test-infra-poc-tekton cluster"
 	rm -rf ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/kubeflow-shared-test-infra-poc-tekton/namespaces/tekton-templates/*
+	mkdir -p ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/kubeflow-shared-test-infra-poc-tekton/namespaces/tekton-templates/
 	####
 	# Add Logic to copy yaml files here
 	####
 	echo "Finish generating yaml files for kubeflow-shared-test-infra-poc-tekton cluster"
 
+# Generate YAML files for Optional-Test-Infra GitOps for optional-test-infra-prow cluster
+optional-generate-optional-test-infra-prow:
+	echo ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}
+	echo "Start generating yaml files for optional-test-infra-prow cluster"
+	rm -rf ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/optional-test-infra-prow/namespaces/prow/*
+	mkdir -p ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/optional-test-infra-prow/namespaces/prow/
+	kubectl -n prow --kubeconfig=./testing/fake-kubeconfig/config.yaml create configmap config \
+	--from-file=config.yaml=./User/clusters/optional-test-infra-prow/namespaces/prow/configmap/config.yaml \
+	--dry-run=client -o yaml > \
+	${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/optional-test-infra-prow/namespaces/prow/config.yaml
+	kubectl -n prow --kubeconfig=./testing/fake-kubeconfig/config.yaml create configmap plugins \
+	--from-file=plugins.yaml=./User/clusters/optional-test-infra-prow/namespaces/prow/configmap/plugins.yaml \
+	--dry-run=client -o yaml > \
+	${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/optional-test-infra-prow/namespaces/prow/plugins.yaml
+	echo "Finish generating yaml files for optional-test-infra-prow cluster"
 
+# Generate YAML files for Optional-Test-Infra GitOps for optional-test-infra-argo cluster
+optional-generate-optional-test-infra-argo:
+	echo ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}
+	echo "Start generating yaml files for optional-test-infra-argo cluster"
+	rm -rf ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/optional-test-infra-argo/namespaces/argo/*
+	mkdir -p ${OPTIONAL_TEST_INFRA_GITOPS_DIRS}/clusters/optional-test-infra-argo/namespaces/argo/
+	####
+	# Add Logic to copy yaml files here
+	####
+	echo "Finish generating yaml files for optional-test-infra-argo cluster"
 
+# Generate YAML files for Optional-Test-Infra GitOps for optional-test-infra-tekton cluster
+optional-generate-optional-test-infra-tekton:
```

Comment:
Added instructions for the new tekton cluster in the [Apply for Shared Test-infra Access](https://docs.google.com/document/d/1hrjIDO7UQ8l5uvADSxbiyM2GDfst1L7vyIKekrrdlwA/edit#) guide, we also need to add the relevant policies in the editor/viewer roles.

Reply:
Thanks, this is ongoing work and should be ready when we announce migration of optional-test-infra from POC to Graduate.

And I also propose we should do IAM as Code way to help external users add their account with permission in automatic way https://github.com/kubeflow/testing/issues/848